DOCS, BUG: Using and clarifying the `request` fixture.

<!--
Thanks for submitting an issue!

Quick check-list while reporting bugs:


-->

Hey all,

I have two queries about the `request` fixture. First one could be a possible documentation change, second is either a bug or my ignorance.

1. I noticed while working with [indirect parametrization](https://docs.pytest.org/en/stable/example/parametrize.html#indirect-parametrization) that the docs assume that the user knows `request` is a fixture and that it is the only parameter name allowed. I only found this out by reading a stack overflow comment ([under accepted answer](https://stackoverflow.com/questions/35413134/indirect-true-vs-indirect-false-in-pytest-mark-parametrize)). Perhaps [in the docs here](https://docs.pytest.org/en/stable/example/parametrize.html#deferring-the-setup-of-parametrized-resources) one could state that `request` is a fixture.

2. Can one cleanly use a fixture for both parametrized and unparametrized tests, if the fixture includes `request` for use with indirect parametrization? The issue I am having is that I have a fixture that is creating some dummy data, and I want to use this dummy data for simple tests and also parametrized tests. What I reaqlly want I guess is a way to cleanly have `request` to have a default value. Code example from the pytest docs:

   ```python
   import pytest


   @pytest.fixture
   def fixt(request):
       return request.param * 3


   @pytest.mark.parametrize("fixt", ["a", "b"], indirect=True)
   def test_indirect(fixt):
       assert len(fixt) == 3


   def test_direct(fixt):  # fails because request.param doesn't exist
       assert len(fixt) == 3
   ```

   Traceback:
   ```bash
   test setup failed
   request = <SubRequest 'fixt' for <Function test_direct>>

       @pytest.fixture
       def fixt(request):
   >       return request.param * 3
   E       AttributeError: 'SubRequest' object has no attribute 'param'
   ```

   My current fix is to add this check to the fixture:

   ```python
   @pytest.fixture
   def fixt(request):
       if hasattr(request, "param"):
           p = request.param
       else:
           p = "x"  # this is my default value
       return p * 3
   ```


### pytest and operating system versions

pytest 8.3.4
OS: Microsoft Windows 10

--- a/changelog/13155.doc.rst
+++ b/changelog/13155.doc.rst
@@ -1,0 +1,1 @@
+Clarified how the ``request`` fixture provides indirect parametrization values via ``request.param``.

--- a/doc/en/example/parametrize.rst
+++ b/doc/en/example/parametrize.rst
@@ -370,6 +370,11 @@ test:
     def test_indirect(fixt):
         assert len(fixt) == 3
 
+The ``request`` argument used by the fixture is pytest's built-in
+:py:class:`FixtureRequest <pytest.FixtureRequest>` fixture. For indirect
+parametrization, the value supplied to the test parameter is passed to the
+fixture and made available as ``request.param``.
+
 This can be used, for example, to do more expensive setup at test run time in
 the fixture, rather than having to run those setup steps at collection time.
 

--- a/doc/en/example/parametrize.rst
+++ b/doc/en/example/parametrize.rst
@@ -370,13 +370,18 @@ test:
     def test_indirect(fixt):
         assert len(fixt) == 3
 
-The ``request`` argument used by the fixture is pytest's built-in
-:py:class:`FixtureRequest <pytest.FixtureRequest>` fixture. For indirect
-parametrization, the value supplied to the test parameter is passed to the
-fixture and made available as ``request.param``.
 
 This can be used, for example, to do more expensive setup at test run time in
 the fixture, rather than having to run those setup steps at collection time.
+
+.. note::
+
+    The ``request`` argument used by the fixture is pytest's built-in
+    :py:class:`FixtureRequest <pytest.FixtureRequest>` fixture. For indirect
+    parametrization, the value supplied to the test parameter is passed to the
+    fixture and made available as ``request.param``.
+
+    For more information, see :ref:`fixture-parametrize`.
 
 .. regendoc:wipe
 

